test/extended/cluster/cl.go

```diff
@@ -202,6 +208,19 @@ func newProject(nsName string) *projectapi.Project {
 	}
 }
 
+func labelNamespace(c kclientset.Interface, ns string, pairs map[string]string) (*v1.Namespace, error) {
+	if len(pairs) == 0 {
+		return nil, nil
+	}
+	var d struct{ Metadata metav1.ObjectMeta `json:"metadata"`}
+	d.Metadata = metav1.ObjectMeta{Labels:pairs}
+	bytes,err := json.Marshal(d)
+	if err != nil {
+		return nil, err
+	}
+	return c.CoreV1().Namespaces().Patch(ns, types.MergePatchType, bytes)
```

Comment:
An easy way to do this with native objects would be to get the existing namespace pointer with:
```
CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
```
Modify the `Namespace.Label` directly by appending or setting the value to `pairs`.

Then just update the namespace:
```
CoreV1().Namespaces().Update(Namespace)
```
where `Namespace` is the modified object you recieved from the first `Get` and which you updated with `pairs`.

Reply:
Done.
With a little concern/question:
Update method uses the whole ns object as the posting body while patch method uses only the label/metadata part. Probably the latter is a bit more efficient.

I tried "oc label project ... --loglelvel=10" and saw only metadata is in the body of http-req.
That was why I used patch method in the first commit here.

Maybe namespaces/projects are not that big, specially for the ones we have just created.
We do not need to worry about that.